DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/DelegateProgress/LearningLogViewModel.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.DelegateProgress
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Models;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+
+    public class LearningLogViewModel : BaseSearchablePageViewModel
+    {
+        public LearningLogViewModel(
+            DelegateProgressAccessRoute accessedVia,
+            LearningLog learningLog,
+            string sortBy,
+            string sortDirection
+        ) : base(null, 1, false, sortBy, sortDirection)
+        {
+            AccessedVia = accessedVia;
+            ProgressId = learningLog.DelegateCourseInfo.ProgressId;
+            CustomisationId = learningLog.DelegateCourseInfo.CustomisationId;
+            CourseName = learningLog.DelegateCourseInfo.CourseName;
+            DelegateId = learningLog.DelegateCourseInfo.DelegateId;
+            DelegateFirstName = learningLog.DelegateCourseInfo.DelegateFirstName;
+            DelegateLastName = learningLog.DelegateCourseInfo.DelegateLastName;
+            DelegateEmail = learningLog.DelegateCourseInfo.DelegateEmail;
+
+            var sortedItems = SortItems(learningLog.Entries);
+
+            Entries = sortedItems.Select(entry => new LearningLogEntryViewModel(entry));
+        }
+
+        public DelegateProgressAccessRoute AccessedVia { get; set; }
+        public int ProgressId { get; set; }
+        public int CustomisationId { get; set; }
+        public string CourseName { get; set; }
+        public int DelegateId { get; set; }
+        public string? DelegateFirstName { get; set; }
+        public string DelegateLastName { get; set; }
+        public string? DelegateEmail { get; set; }
+        public IEnumerable<LearningLogEntryViewModel> Entries { get; set; }
+
+        public string DelegateFullName =>

```

Comment:
This hasn't been added to the ticket, but we should be showing names as `LastName, FirstName` anywhere we sort by them.

Reply:
I had changed this to use `LastName, FirstName` but then realised we don't sort by the name here. I've changed it back, but left the new string display method in the helper. Do we want to get rid of that as well or leave it for whenever we fix the rest of the names?

Reply:
Hmm, I think we really want two methods - one for sortable names and one for others.
If we don't sort by this name it should be the standard `Forename Surname`.
Feel free to use a helper method, but could you make it clear whether it's for sortable names or non-sortable names? Maybe there's a better adjective to use than that.